modules/rostests/apitests/user32/GetDCEx.c

```diff
@@ -466,16 +463,746 @@ Test_GetDCEx_CS_SwitchedStyle()
 
     DestroyWindow(hwnd1);
     DestroyWindow(hwnd2);
-    ok(UnregisterClass(pszClassName, GetModuleHandleA(0)) == TRUE,
+    ok(UnregisterClassW(pszClassName, GetModuleHandleW(0)) == TRUE,
        "UnregisterClass failed\n");
 }
 
+/* test behavior of DC attributes with various GetDC/ReleaseDC combinations */
```

Comment:
Are all these tests that follow below those that "come from wine"?

Reply:
Yes, the tests all follow "test behavior of DC attributes" after the ones that "come from wine"(winetests/user32/dce.c). I tried to insert them using include, but the wine code has differences that prevent compilation. And I don't know if it's a good idea to modify the wine code for future synchronization. I understand that duplicate code is not ideal, but somehow we should add these functions to rostests.